lib/SIL/SILBasicBlock.cpp

```diff
@@ -145,112 +146,54 @@ void SILBasicBlock::cloneArgumentList(SILBasicBlock *Other) {
 SILFunctionArgument *
 SILBasicBlock::createFunctionArgument(SILType Ty, const ValueDecl *D,
                                       bool disableEntryBlockVerification) {
-  assert((disableEntryBlockVerification || isEntry()) &&
-         "Function Arguments can only be in the entry block");
-  const SILFunction *Parent = getParent();
   auto OwnershipKind = ValueOwnershipKind(
       *Parent, Ty,
       Parent->getConventions().getSILArgumentConvention(getNumArguments()));
-  return new (getModule()) SILFunctionArgument(this, Ty, OwnershipKind, D);
```

Comment:
I don't know why `createFunctionArgument` is a `SILBasicBlock` API in the first place as opposed to `SILFunction`, but what's the purpose of moving it to a builder?

Reply:
Ok, I guess the purpose of consolidating the function/phi argument helper is to centralize some replace/insert argument code.